app/assets/javascripts/analytics.js

```diff
@@ -0,0 +1,209 @@
+"use strict";
+
+class AnalyticsDashboard {
+  constructor () {
+    this.barPadding   = 1;
+    this.bottomMargin = 100;
+    this.margins      = { bottom: 30 };
+    this.xAxis        = true;
+    this.labels       = true;
+  }
+
+  render () {
+    this.svg
+      .attr("width", this.width)
+      .attr("height", this.height + this.margins.bottom);
+
+    this.setYScale(this.data);
+    this.draw();
+
+    if(this.labels) {
+      this.drawLabels();
+    }
+
+    if(this.xAxis) {
+      this.drawAxis();
+    }
+  }
+
+  update () {
+    this.setYScale(this.data);
+
+    this.svg.selectAll(".bar")
+      .data(this.data)
+      .transition()
+      .duration(750)
+      .attr("height", (d) => {
+        return this.scale(d.value);
+      })
+      .attr("y", (d) => {
+        return this.height - this.scale(d.value);
+      });
+
+    this.drawLabels();
+
+    this.svg.selectAll(".label")
+      .data(this.data)
+      .transition()
+      .duration(1000)
+      .text((d) => {
+        return d.value;
+      })
+      .attr("y",    this.setYForLabel.bind(this)    )
+      .attr("fill", this.setFillForLabel.bind(this) );
+  }
+
+  setYForLabel (d) {
+    let scaled  = this.scale(d.value),
+        y       = this.height - scaled + 15;
+
+    if(scaled < 20 ) {
```

Comment:
that makes sense. you're using it in a few places so it would be great to name it `MIN_FOR_LABEL_INSIDE_BAR` or something